project/Settings.scala

```diff
@@ -109,42 +109,57 @@ object Settings {
+    scalacOptions in(Compile, console) --= consoleHostileSettings,
+    addCompilerPlugin("org.scalamacros" % "paradise" % paradiseV cross CrossVersion.full)
   )
 
-  val docSettings = List(
-    // http://stackoverflow.com/questions/31488335/scaladoc-2-11-6-fails-on-throws-tag-with-unable-to-find-any-member-to-link#31497874
-    "-no-link-warnings"
+  val crossVersionSettings = List(
+    crossScalaVersions := List(ScalaVersion212, ScalaVersion211)
   )
 
+  val dockerTags = settingKey[Seq[String]]("The tags for docker builds.")
+
   lazy val dockerSettings = Seq(
-    imageNames in docker := Seq(
-      ImageName(
-        namespace = Option("broadinstitute"),
-        repository = name.value,
-        tag = Option(cromwellVersion)),
-      ImageName(
-        namespace = Option("broadinstitute"),
-        repository = name.value,
-        tag = Option(version.value))
-    ),
+    /*
+    NOTE: Gave up fighting with SBT settings. Using an environment variable instead.
+
+    The below "just works", assuming womtool docker image building is also enabled, setting the right image names and
+    versions.
+
+    `sbt 'show docker::imageNames'` returns:
+      ArrayBuffer(broadinstitute/womtool:30, broadinstitute/womtool:30-c33be41-SNAP)
+      ArrayBuffer(broadinstitute/cromwell:30, broadinstitute/cromwell:30-c33be41-SNAP)
+
+    `CROMWELL_DOCKER_TAGS=dev,develop sbt 'show docker::imageNames'` returns:
+      ArrayBuffer(broadinstitute/womtool:dev, broadinstitute/womtool:develop)
+      ArrayBuffer(broadinstitute/cromwell:dev, broadinstitute/cromwell:develop)
+    */
+    dockerTags := sys.env.getOrElse("CROMWELL_DOCKER_TAGS", s"$cromwellVersion,${version.value}").split(","),
+    imageNames in docker := dockerTags.value map { tag =>
+      ImageName(namespace = Option("broadinstitute"), repository = name.value, tag = Option(tag))
+    },
     dockerfile in docker := {
       // The assembly task generates a fat JAR file
       val artifact: File = assembly.value
       val artifactTargetPath = s"/app/${artifact.name}"
+      val artifactName = name.value
 
       new Dockerfile {
         from("openjdk:8")
         expose(8000)
         add(artifact, artifactTargetPath)
-        runRaw(s"ln -s $artifactTargetPath /app/cromwell.jar")
+        runRaw(s"ln -s $artifactTargetPath /app/$artifactName.jar")
 
         // If you use the 'exec' form for an entry point, shell processing is not performed and
         // environment variable substitution does not occur.  Thus we have to /bin/bash here
         // and pass along any subsequent command line arguments
         // See https://docs.docker.com/engine/reference/builder/#/entrypoint
-        entryPoint("/bin/bash", "-c", "java ${JAVA_OPTS} -jar /app/cromwell.jar ${CROMWELL_ARGS} ${*}", "--")
+        entryPoint(
+          "/bin/bash",
+          "-c",
+          s"java $${JAVA_OPTS} -jar /app/$artifactName.jar $${${artifactName.toUpperCase}_ARGS} $${*}",
```

Comment:
The more stringent compiler flags weren't being applied to Centaur before this PR? 😦

Reply:
The highlighted change specifically allows us to build dockers for any executable within the monorepo, replacing `cromwell` with `$artifactName`. AFAIK, the above isn't related to the compiler flags, but maybe I missed your question.

Reply:
Hmm I thought you had to add the "dollar doubling" here to silence the "fatal warnings" about missing an 's' when the compiler thought you were trying to do interpolations. But perhaps the double dollars were added because you explicitly turned on interpolation to be able to get the artifact name in. 😄